playbooks/deploy_standalone.yml

```
@@ -0,0 +1,138 @@
+
+    - name: Install CLI tools
+      become: true
+      when: install_cli_tools
+      block:
+        - name: Install and configure k9s
+          ansible.builtin.include_role:
+            name: azimuth_cloud.azimuth_ops.k9s
+
+        - name: Get installed Kubernetes version
+          ansible.builtin.command: k3s kubectl version --output json
+          changed_when: false
+          register: k3s_kubectl_version
+
+        - name: Set kubectl version fact
+          ansible.builtin.set_fact:
+            kubectl_version: "{{ (k3s_kubectl_version.stdout | from_json).serverVersion.gitVersion.split('+') | first }}"
+
+        - name: Install Kubectl
+          ansible.builtin.include_role:
+            name: azimuth_cloud.azimuth_ops.kubectl
+
+        - name: Install Helm
+          ansible.builtin.include_role:
+            name: azimuth_cloud.azimuth_ops.helm
+
+        - name: Install Kustomize
+          ansible.builtin.include_role:
+            name: azimuth_cloud.azimuth_ops.kustomize
+
+        - name: Install Flux
+          ansible.builtin.include_role:
+            name: azimuth_cloud.azimuth_ops.flux
+            tasks_from: cli
+          when: flux_enabled
+
+    - name: Setup Kubeconfig
+      when: slurp_k3s_kubeconfig
+      become: true
+      block:
+        - name: Slurp kubeconfig file
+          ansible.builtin.slurp:
+            src: /etc/rancher/k3s/k3s.yaml
+          register: k3s_kubeconfig
+
+        - name: Ensure kube config directory exists
+          ansible.builtin.file:
+            path: "{{ ansible_env.HOME }}/.kube"
+            state: directory
+            mode: u=rwx,g=rx,o=rx
+
+        - name: Write kubeconfig file
+          ansible.builtin.copy:
+            content: "{{ k3s_kubeconfig.content | b64decode }}"
+            dest: "{{ ansible_env.HOME }}/.kube/config"
+            mode: u=rwx,g=r,o=r
+
+    # For a single node install, we put the monitoring and ingress controller on the K3S cluster
+    - name: Install monitoring stack and ingress controller
+      when: install_mode == 'singlenode'
+
+      # Configure the K3S cluster as a Cluster API management cluster when doing a HA installation
+      block:
+        # Must be done before NGINX ingress so that the ServiceMonitor CRD exists
+        - name: Install Kube-Prometheus-Stack
+          ansible.builtin.include_role:
+            name: azimuth_cloud.azimuth_ops.kube_prometheus_stack
+          when: deploy_prometheus_stack | default(false)
+
+        - name: Install Nginx ingress controller
+          ansible.builtin.include_role:
+            name: azimuth_cloud.azimuth_ops.ingress_nginx
+          when: "ingress_controller_enabled | default(true)"
+
+    # HA mode relies on openstack
+    - name: Configure Azimuth HA cluster
+      when: install_mode == 'ha'
+      block:
+        - name: Fail to install HA
+          ansible.builtin.fail:
+            msg: "HA mode without openstack is unsupported"
```

I think we should have a completely separate `install_mode: "standalone"` or something like that for the standalone case so as not to confuse it with the existing `singlenode` and `ha` install modes.

We could then modify this task and move it to the top of this playbook so that it gets executed first. That would allow us to remove all other `when: install_mode == ...` conditionals from other task in this playbook and make the logic simpler to follow.

```suggestion
    # HA mode relies on openstack
    - name: Fail if install_mode is not standalone
      ansible.builtin.fail:
        msg: "Install modes other than 'standalone' are not supported for this playbook"
      when: install_mode != 'standalone'
```

---

Me and will have already talked about the possibility of doing something like that, if your seconding it as a good idea then I'll work on implementing it